kubectl-plugin/pkg/cmd/job/job_submit.go

```diff
@@ -232,10 +269,50 @@ func (options *SubmitJobOptions) Run(ctx context.Context, factory cmdutil.Factor
 		return fmt.Errorf("failed to initialize clientset: %w", err)
 	}
 
-	// createdRayJob, err = k8sClients.CreateRayCustomResource(ctx, util.RayJob, options.configFlags.Namespace, unstructuredRayjob)
-	options.RayJob, err = k8sClients.DynamicClient().Resource(util.RayJobGVR).Namespace(*options.configFlags.Namespace).Create(ctx, options.RayJob, v1.CreateOptions{})
-	if err != nil {
-		return fmt.Errorf("Error when creating RayJob CR: %w", err)
+	if options.fileName == "" {
+		// Genarate the ray job.
+		rayJobObject := generation.RayJobYamlObject{
+			RayJobName:     options.rayjobName,
+			Namespace:      *options.configFlags.Namespace,
+			SubmissionMode: "InteractiveMode",
+			RayClusterSpecObject: generation.RayClusterSpecObject{
+				RayVersion:     options.rayVersion,
+				Image:          options.image,
+				HeadCPU:        options.headCPU,
+				HeadMemory:     options.headMemory,
+				WorkerGrpName:  options.workerGrpName,
+				WorkerCPU:      options.workerCPU,
+				WorkerMemory:   options.workerMemory,
+				WorkerReplicas: options.workerReplicas,
+				// This is here to match the existing rayjob sample.
+				WorkerLifecyclePrestopExecComand: []string{"/bin/sh", "-c", "ray stop"},
+			},
+		}
+		rayJobApplyConfig := rayJobObject.GenerateRayJobApplyConfig()
+
+		// Print out the yaml if it is a dry run
+		if options.dryRun {
+			resultYaml, err := generation.ConvertRayJobApplyConfigToYaml(rayJobApplyConfig)
+			if err != nil {
+				return fmt.Errorf("Failed to convert rayjob into yaml format: %w", err)
+			}
+
+			fmt.Printf("%s\n", resultYaml)
+			return nil
+		}
+
+		// Apply the generated yaml
+		rayJobApplyConfigResult, err := k8sClients.RayClient().RayV1().RayJobs(*options.configFlags.Namespace).Apply(ctx, rayJobApplyConfig, v1.ApplyOptions{FieldManager: "kubectl-plugin"})
```

Comment:
The field manager should be something like `ray-kubectl-plugin`, since `kubectl-plugin` is too generic when viewing the object from the server-side